Adds channel to audio event permitted params (#720)
QutEcoacoustics/baw-server: spec/README.md

Note that all api specs are run as  the `admin_user`.

#### Unpermitted parameter error in API specs

> If you encounter an unpermitted parameter error in your API specs, ensure you
> set `consume` to nil. The default value is 'json' and can cause issues.

Reviewer: ```suggestion
If you encounter an unpermitted parameter error in your API specs, try checking that the 
correct `Content-Type` is being sent to the API.
The default content type for requests in the API is `application/json` but for requests
that don't expect a body you can set `consumes` in the API spec definition to `nil`. 
```

The `factory_bot` gem is used to create test data. Factories are defined in the
`spec/factories` folder.

> `create_entire_hierarchy` is a helper method that creates a complete data
> hierarchy, with all necessary associations (e.g. an audio recording that
> belongs to a site that belongs to a region that belongs to a project).

Reviewer: ```suggestion
`create_entire_hierarchy` is a helper method that creates a complete data
hierarchy, with all necessary associations (e.g. an audio recording that
belongs to a site that belongs to a region that belongs to a project).
It also makes a hierarchy that makes _sense_: all the relationships
are semantically correct, E.g. the same user is used to create many of the entities,
or multiple references to a project all reference the same parent project.
```

> Factory cascading creation may result in unneccesary branches which could slow
> down tests. This can be avoided by specifying a value for the assocation upon
> creation.

Reviewer: ````suggestion
Watch out for standard factory bot factories. Each time a relation is needed, a 
**new** instance of that entity is created. Entities created from previous relation
instantiations are generally **not reused**.
This results in explosive/exponential entity creations and slows down tests considerably.
This can be avoided by specifying a value for the relationship upon creation:

```ruby
# this
create(:audio_recording, site: an_existing_site, creator: an_existing_user)
# instead of
create(:audio_recording)
```

...or by using the `create_entire_hierarchy` helper described above.
````

E.g. The items that will be created using the `audio_recording` factory with no
arguments:

                            audio_recording
                            /             \
                          site             user
                       /    |    \
                      /     |      user
                region    project
               /      \          \
              user     \          user
                     project
                         \
                        user

> 

Reviewer: this is great, but take advantage of GH built in support for mermaid diagrams. I reckon and ERD is the way to go: https://mermaid.js.org/syntax/entityRelationshipDiagram.html

Also do a compare and contrast of `create_entire_hierarchy` vs `create` with the diagrams (add a second diagram)

or multiple references to a project all reference the same parent project.

Watch out for standard factory bot factories. Each time a relation is needed, a
**new** instance of that entity is created. Entities created from previous
relation instantiations are generally **not reused**.

This results in explosive/exponential entity creations and slows down tests
considerably. This can be avoided by specifying a value for the relationship
upon creation:

```ruby
# this
create(:audio_recording, site: an_existing_site, creator: an_existing_user)
# instead of
create(:audio_recording)
```

...or by using the `create_entire_hierarchy` helper described above.

E.g. The items that will be created using the `audio_recording` factory with no
arguments:

```mermaid
erDiagram
    user1["USER"] { int id "1" }
    user2["USER"] { int id "2" }
    user3["USER"] { int id "3" }
    user4["USER"] { int id "4" }
    user5["USER"] { int id "5" }
    project1["PROJECT"] { int id "1" }
    project2["PROJECT"] { int id "2" }

    AUDIO_RECORDING }|--|| user1 : "creator"
    AUDIO_RECORDING }|--|| SITE : ""

    SITE }|--|| user2 : "creator"
    SITE }|--|| REGION : ""

>     SITE }|--|| project1 : ""

Reviewer: ```suggestion
    SITE }|--|{ project1 : ""
```

    project1 }|--|| user3 : "creator"

    REGION }|--|| user4 : "creator"
    REGION }|--|| project2 : ""

    project2 }|--|| user5 : "creator"

```

Compared to the entities generarted by the `create_entire_hierarchy` helper that
are relevant to an `audio_recording`:

```mermaid
erDiagram
user["USER"] { int id "1" }
project["PROJECT"] { int id "1" }

    AUDIO_RECORDING }|--|| user : "creator"
    AUDIO_RECORDING }|--|| SITE : ""

    SITE }|--|| user : "creator"
    SITE }|--|| REGION : ""

>     SITE }|--|| project : ""

Reviewer: ```suggestion
    SITE }|--|{ project : ""
```
